var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -96,20 +83,20 @@ class Dealii(CMakePackage):
 
     # optional dependencies
     depends_on("mpi",              when="+mpi")
-    depends_on("arpack-ng+mpi",    when='+arpack+mpi')
```

Comment:
I was confused about this as well. So, according to the way it used to be, if I build `+arpack~mpi` it doesn't build with arpack, only when I build with `+arpack+mpi`? That doesn't seem... ideal. What if we raised an error when someone tries to build `+arpack~mpi`?

In hindsight, this wasn't the change I should have made. I should have removed `arpack-ng+mpi`, but not `+arpack+mpi`.